jest/src/test/java/io/searchbox/client/FailingProxy.java

```diff
@@ -0,0 +1,112 @@
+package io.searchbox.client;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Inet4Address;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+import org.apache.commons.lang3.StringEscapeUtils;
+import org.slf4j.Logger;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * A really basic http server that responds with a 500 error and some text/html.
+ */
+class FailingProxy implements Runnable {
```

Comment:
If possible please use `DefaultHttpProxyServer` from littleproxy instead, we are already using it for `JestHttpClientConfiguredProxyIntegrationTest` (as an example).